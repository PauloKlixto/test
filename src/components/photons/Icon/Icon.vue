<template>
  <img class="Icon" v-bind="$attrs" :src="src" @click="$emit('click')" />
</template>

<script>
export default {
  name: 'Icon',
  props: {
    name: {
      type: String,
      required: true
    },
    path: {
      type: String,
      default: 'icons'
    }
  },
  computed: {
    src() {
      return require(`@/assets/${this.path}/${this.name}.svg`)
    }
  }
}
</script>
