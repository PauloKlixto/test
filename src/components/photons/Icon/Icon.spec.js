/* eslint-disable no-undef */
import { shallowMount } from '@vue/test-utils'
import Icon from './Icon'

describe('Icon', () => {
  const WRAPPER = shallowMount(Icon, {
    propsData: {
      name: 'check'
    }
  })

  test('Componente existe', () => {
    expect(WRAPPER.exists()).toBe(true)
  })

  test('Possui as classes necessárias', () => {
    expect(WRAPPER.find('img').classes()).toContain('Icon')
  })

  test('Contém o svg na tag img', () => {
    expect(WRAPPER.find('img').attributes().src).toEqual('check.svg')
  })

  test('Componente não alterado', () => {
    expect(WRAPPER.html()).toMatchSnapshot()
  })
})
