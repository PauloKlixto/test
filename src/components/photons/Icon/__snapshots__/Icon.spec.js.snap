// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Componente n√£o alterado 1`] = `"<img src=\\"check.svg\\" class=\\"Icon\\">"`;
